app/main.py

```diff
@@ -1,8 +1,22 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> dict:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+
+    for person in people:
+        Person(person["name"], person["age"])
+    for person in people:
+        individual = Person.people[person["name"]]
+        if person.get("wife") and person["wife"] is not None:
+            individual.__dict__["wife"] = Person.people[person["wife"]]
+        if person.get("husband") and person["husband"] is not None:
+            individual.__dict__["husband"] = Person.people[person["husband"]]
+        person_list.append(Person.people[person["name"]])
```

Comment:
why do you do this append here? It is more logical to make in the first loop `$13-14` lines